### ISSUE TITLE
Can't build for MacOS (Latest)

### ISSUE THREAD

**Issue by patientplatypus6** [servo/mozjs#329]
Can't build for macOS (latest OS - Ventura). The following error message occurs (can't build crate `https://github.com/servo/mozjs`)

```
error: failed to run custom build command for `mozjs_sys v0.68.2 (https://github.com/servo/mozjs#f87705d6)`

Caused by:
  process didn't exit successfully: `/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-e6c18cb04a3d67dd/build-script-build` (exit status: 101)
  --- stdout
  cargo:outdir=/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-c9599b8984f352e1/out/build
  [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure.in ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure || true
  [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure.in ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure || true
  ! [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure.in -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure || true
  ! [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure.in -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure || true
  if [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure -nt config.status ]] ; then \
  	  PYTHON="python2" MOZ_TOOLS="" \
  	  CC="clang" CFLAGS="" \
  	  CPP="gcc -E" CPPFLAGS="-I/Users/peterweyand/.cargo/registry/src/github.com-1ecc6299db9ec823/encoding_c_mem-0.2.6/include " \
  	  CXX="clang++" CXXFLAGS="-stdlib=libc++" \
  	  AS="" AR="ar" \
  	  STLPORT_LIBS="" \
  	  '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --disable-export-js --disable-shared-js --build-backends=RecursiveMake --enable-js-streams || (cat config.log && exit 1) ; \
  	fi
  created virtual environment CPython3.10.8.final.0-64 in 18ms
    creator CPython3Posix(dest=/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-c9599b8984f352e1/out/build/_virtualenvs/build, clear=False, no_vcs_ignore=False, global=False)
    activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator
  Using Python 3.10.8 from /Users/peterweyand/Code/servo/python/_virtualenv3.10/bin/python3
  Site not up-to-date reason: "/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-c9599b8984f352e1/out/build/_virtualenvs/build" does not exist

  --- stderr
  Traceback (most recent call last):
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/js/src/../../configure.py", line 350, in <module>
      sys.exit(main(sys.argv))
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/js/src/../../configure.py", line 121, in main
      _activate_build_virtualenv()
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/js/src/../../configure.py", line 344, in _activate_build_virtualenv
      if not build_site.ensure():
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/python/mach/mach/site.py", line 565, in ensure
      _create_venv_with_pthfile(
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/python/mach/mach/site.py", line 1191, in _create_venv_with_pthfile
      with open(os.path.join(platlib_site_packages_dir, PTH_FILENAME), "w") as f:
  FileNotFoundError: [Errno 2] No such file or directory: '/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-c9599b8984f352e1/out/build/_virtualenvs/build/lib/python3.10/site-packages/mach.pth'
  cat: config.log: No such file or directory
  make: *** [maybe-configure] Error 1
  thread 'main' panicked at 'assertion failed: result.success()', /Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/build.rs:178:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
      Timing report saved to /Users/peterweyand/Code/servo/target/cargo-timings/cargo-timing-20221215T002819Z.html
[Warning] Could not generate notification! 
Build FAILED in 0:01:28
```

**Comment by patientplatypus6** [comment]
I've gone ahead and done a build clean and build clean-uwp, but the error keeps occurring. From what I can tell the crates aren't reinstalling on new builds, it's just checking that they exist (which they did from a prior installation). Someone mentioned that it might have something to do with the ordering of the crates in the TOML file? Thanks.

**Comment by coreyja** [comment]
Was in Discord with @patientplatypus6 and tried to compile as well and got a very similiar error

```
FileNotFoundError: [Errno 2] No such file or directory: '/Users/coreyja/src/mozjs/target/debug/build/mozjs_sys-a86259d36c1dbf46/out/build/_virtualenvs/build/lib/python3.10/site-packages/mach.pth'
```

OS Version Info in case it helps
```
✦ [13:17]  sw_vers
ProductName:            macOS
ProductVersion:         13.0.1
BuildVersion:           22A400
```

**Comment by patientplatypus6** [comment]
And mine - 

```
peterweyand@Peters-MacBook-Pro mozjs % sw_vers
ProductName:		macOS
ProductVersion:		13.1
BuildVersion:		22C65
```

**Comment by patientplatypus6** [comment]
Found the issue - https://bugzilla.mozilla.org/show_bug.cgi?id=1766497

The hacky solution is to uninstall `python@3.10` and install `python@3.9`. In python@3.10 `venv` is used instead of `virtualenv` which has been dropped as deprecated and this package uses `virtualenv` throughout. Consider adding a flag to a PR that checks python version and then changes supported keyword useage. Don't know how annoying this would be, but moving forward using the deprecated syntax will cause all future versions of python to crash, making this package, and Servo (and other packages that use this package) non-backwards compatible. Thanks.

**Comment by ales-tsurko** [comment]
In case of homebrew, you can keep your existing version of python and install `python 3.9` with `brew install python3.9`. Then you just set `PYTHON3` variable before you build:

```
$ PYTHON3=$(which python3.9) ./mach build --dev
```

**Comment by mrobinson** [comment]
I believe this is fixed by #358.
